File: backend/src/main/java/com/ufc/backend/backend/model/Semester.java
package com.ufc.backend.backend.model;

import lombok.*;
import org.springframework.data.mongodb.core.mapping.DBRef;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Semester {
    private Integer index;
    @DBRef
    private Set<Classes> classes = new HashSet<>();
    public boolean equals(Semester obj) {
        return this.getIndex().equals(obj.getIndex());
    }
}

File: backend/src/main/java/com/ufc/backend/backend/model/Semester.kt
package com.ufc.backend.backend.model


data class Semester
    (
    var index: Int,
    var subject: MutableSet<Subject>? = null
) {
    val actualSemesterIndex = this.index - 1
    override fun equals(other: Any?): Boolean {
        return if (javaClass != other!!.javaClass) {
            false
        } else this.index == (other as Semester).index
    }

    override fun hashCode(): Int {
        return index
    }
}